modelator/utils/modelator_helpers.py

```diff
@@ -2,6 +2,36 @@
 from typing import Dict
 from .. import const_values
 import logging
+import subprocess
+from urllib.request import urlopen
+import zipfile
+import io
+
+
+def check_for_apalache_jar() -> bool:
+    # check for existence of the correct Apalache jar.
+    # If it is missing, download it to const_value.DEFAULT_APALACHE_LOCATION and return False.
+    # If it is already there, do nothing and return True.
+    try:
+        download_needed = False
+        out = subprocess.run(
+            ["java", "-jar", const_values.DEFAULT_APALACHE_JAR, "version"],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        version = out.stdout.decode("utf-8").strip()
+        if not version == const_values.DEFAULT_APALACHE_VERSION:
```

Comment:
~We should also verify the integrity checksum to make sure the downloaded JAR is not tampered.~

Sorry, the integrity check should be done when the file is downloaded, not here.